var/spack/repos/builtin/packages/py-ipywidgets/package.py

```diff
@@ -12,9 +12,17 @@ class PyIpywidgets(PythonPackage):
     homepage = "https://github.com/ipython/ipywidgets"
     url      = "https://github.com/ipython/ipywidgets/archive/5.2.2.tar.gz"
 
+    version('7.4.2', sha256='f156165e8a855ed862fdf48e72700bdcd6956d089a2018c5b36d358255d45b2b')
     version('5.2.2', sha256='d61ab8bb12b90981a3a6010429816d70eaa041e622043207bcb74239b664d4f3')
 
-    depends_on('python@2.7:2.8,3.3:')
-    depends_on('py-ipython@4.0.0:', type=('build', 'run'))
+    depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
+    depends_on('py-ipython@4:', type=('build', 'run'))
+    depends_on('py-ipython@4:5', type=('build', 'run'), when='^python@:3.2')
     depends_on('py-ipykernel@4.2.2:', type=('build', 'run'))
+    depends_on('py-ipykernel@4.5.1:', type=('build', 'run'), when='@6:')
     depends_on('py-traitlets@4.2.1:', type=('build', 'run'))
+    depends_on('py-traitlets@4.3.1:', type=('build', 'run'), when='@6:')
+    depends_on('py-nbformat@4.2.0:', type=('build', 'run'), when='@6:')
+    depends_on('py-widgetsnbextension@3.4.0:3.4.999', type=('build', 'run'), when='@6:')
```

Comment:
Is there a PR to update py-widgetsnbextension?

Reply:
Now there is: #15105

Reply:
#15105 is for version 3.3.0 but the dependency above is for 3.4.

Reply:
Added 3.5.1

Reply:
You will need 3.4.x to match the dependency specified for ipywidgets.
```
depends_on('py-widgetsnbextension@3.4.0:3.4.999', type=('build', 'run'), when='@6:')
```

Reply:
And 3.4.2